web-registry/src/features/marketplace/BuySellOrderFlow/hooks/useFetchSellOrders.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useMemo } from 'react';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+
+import { useLedger } from 'ledger';
+import {
+  getSellOrdersExtentedQuery,
+  SELL_ORDERS_EXTENTED_KEY,
+} from 'lib/queries/react-query/marketplace/getSellOrdersExtentedQuery/getSellOrdersExtentedQuery';
+
+import { SellOrderInfoExtented } from 'hooks/useQuerySellOrders';
+
+type UseFetchSellOrdersResponse = {
+  sellOrders: SellOrderInfoExtented[] | undefined;
+  refetchSellOrders: () => Promise<SellOrderInfoExtented[] | undefined>;
+};
+
+export const useFetchSellOrders = (): UseFetchSellOrdersResponse => {
+  const { marketplaceClient } = useLedger();
+  const reactQueryClient = useQueryClient();
+
+  // Query
+  const sellOrdersQuery = useMemo(
+    () =>
+      getSellOrdersExtentedQuery({
+        enabled: !!marketplaceClient,
+        client: marketplaceClient,
+        reactQueryClient,
+        request: {},
+      }),
+    [marketplaceClient, reactQueryClient],
+  );
+
+  // Fetch
+  const { data: sellOrders } = useQuery(sellOrdersQuery);
+
+  // Refetch callback
+  const refetchSellOrders = useCallback(async (): Promise<
+    SellOrderInfoExtented[] | undefined
+  > => {
+    await reactQueryClient.invalidateQueries({
```

Comment:
Here invalidating the cache will automatically run again all query using this key.
Below we're returning the new result from the cache because this callback needs to return the new value but in most cases reload callbacks just need to invalidate a key is enough to refresh data and the UI.

Reply:
I think we could add that as comment in the code itself